app/javascript/components/CompanyTable.tsx

```diff
@@ -44,12 +44,12 @@ interface CompanyTableProps{
 const CompanyTable: React.FC <CompanyTableProps> = (props: CompanyTableProps) => {
    const {companies, setCompany} = props;
   React.useEffect(() => {
-    axios.get('/companies.json').then((response) => {
+    httpClients.companies.get_data().then((response) => {
       setCompany(response.data);
     });
   }, []);
```

Comment:
U can add 'companies' to dependencies of useEffect (in square brackets) & combine two lines (47 and 48)